### ISSUE TITLE
Webstorm is showing error because can't import webpack.prod.babel.js

### ISSUE THREAD

**Issue by Surm4** [Can-Sahin/react-boilerplate-typescript#67]
## Description
Webstorm is showing error when I open just installed project which says cannot import module 'webpack.prod.babel.js'

## Steps to reproduce
- clone boilerplate
- npm run setup
- npm run clean
- all Loadable files are marked as error

(Add link to a demo on https://jsfiddle.net or similar if possible)
--

**Expected behavior**
No error

**Screenshots**
![Imgur](https://imgur.com/7LI5uKe.png)

## Versions

- React-Boilerplate:
- Node/NPM: 12.6.1 on nvm/ 6.13.4
- Browser: Chrome Canary 84.0.4104.0 

## Additional Info:
I have no error and everything works for me when I've added this lines to eslintrc.js
But I didn't test it deeply
![image](https://user-images.githubusercontent.com/40576160/78453869-d2acd180-7694-11ea-8104-d02c411cd34a.png)

**Comment by gjolund** [comment]
I am also experiencing this issue when running `npm run lint` directly from the command line.

I have noticed that the module resolution is incorrect as it is routing `internals` into `node_modules/react-redux`.

I have also noticed that this issue begins to manifest when you have multiple copies of `react-boilerplate-typescript` in the same project.

For instance I am using [lerna](https://lerna.js.org/) to manage my monorepo, but I have been able to reproduce the error by creating a repo and cloning `react-boilerplate-typescript` into two sub directories and removing their `.git` directories.

It also seems to manifest when sharing a project directory with any other node app that has `node_modules`.

I haven't had the time to dig too deeply into this, and have managed to work around it for now, but I am afraid that it is a sign of a deeper issue with dependency linking and module resolution.

Any insight into this would be very appreciated.

```
code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/home/austin/Workspace/Sophia/sophia-iot/frontend/node_modules/eslint-import-resolver-webpack/index.js',
    '/home/austin/Workspace/Sophia/sophia-iot/frontend/node_modules/eslint-module-utils/resolve.js',
    '/home/austin/Workspace/Sophia/sophia-iot/frontend/node_modules/eslint-plugin-import/lib/rules/no-unresolved.js',
    '/home/austin/Workspace/Sophia/sophia-iot/frontend/node_modules/eslint-plugin-import/lib/index.js',
    '/home/austin/Workspace/Sophia/sophia-iot/frontend/node_modules/eslint/lib/cli-engine/config-array-factory.js',
    '/home/austin/Workspace/Sophia/sophia-iot/frontend/node_modules/eslint/lib/cli-engine/cascading-config-array-factory.js',
    '/home/austin/Workspace/Sophia/sophia-iot/frontend/node_modules/eslint/lib/cli-engine/cli-engine.js',
    '/home/austin/Workspace/Sophia/sophia-iot/frontend/node_modules/eslint/lib/cli-engine/index.js',
    '/home/austin/Workspace/Sophia/sophia-iot/frontend/node_modules/eslint/lib/cli.js',
    '/home/austin/Workspace/Sophia/sophia-iot/frontend/node_modules/eslint/bin/eslint.js'
  ]
}
Error: Cannot find module '/home/austin/Workspace/Sophia/sophia-iot/frontend/node_modules/react-redux/internals/webpack/webpack.prod.babel.js'
```

**Comment by gjolund** [comment]
@SurmaAa adding `node.extensions` to my settings seems to have resolved the issue for me also.

My updated `eslintrc.js` `settings` object now looks like this:

```
settings: {
    'import/parsers': {
      '@typescript-eslint/parser': ['.ts', '.tsx'],
    },
    'import/resolver': {
      node: {
        extensions: ['.js', '.jsx', '.ts', '.tsx']
      },
      webpack: {
        config: './internals/webpack/webpack.prod.babel.js',
      },
      typescript: {
        alwaysTryTypes: true, // always try to resolve types under `<root>@types` directory even it doesn't contain any source code, like `@types/unist`
        directory: './tsconfig.json',
      },
    },
    'import/ignore': ['types'], // Weirdly eslint cannot resolve exports in types folder (try removing this later)
  },
```

```
```

**Comment by Can-Sahin** [comment]
Hmm interesting. @SurmaAa does `npm run lint` works fine for you? If it does then I think its related to eslint in webstorm, because I am using vscode and had no problem. 

Adding `ts` files to node extensions is a fix it seems. `eslint-import-resolver-typescript` is used and I thought that does the trick for finding `ts` files. I'll look into this.

**Comment by idMolotov** [comment]
confirm this issue.


It looks like some mapping/names overrides is happens here.


I've make a fix in the `app/styles/styled-components.ts` file.

Changing import from 
```
- import * as styledComponents from 'styled-components';
+ import * as styledComponents from 'styled-components/index';
```

Eslint error message is disappears.

**Comment by Surm4** [comment]
> Hmm interesting. @SurmaAa does `npm run lint` works fine for you? If it does then I think its related to eslint in webstorm, because I am using vscode and had no problem.
> 
> Adding `ts` files to node extensions is a fix it seems. `eslint-import-resolver-typescript` is used and I thought that does the trick for finding `ts` files. I'll look into this.

I didn't check because I installed eslint-loader and it works fine. 
However I checked `npm run lint` and it failed on `lint:css` so I changed script line to `"stylelint \"./app/**/*.{js,ts,tsx}\""` but now it fails on `"lint:eslint:quiet` what's strange I have proper errors from es-lint loader in browser and there it's not failing
![Imgur](https://imgur.com/cGuGI7b.png)

**Comment by Surm4** [comment]
> > Hmm interesting. @SurmaAa does `npm run lint` works fine for you? If it does then I think its related to eslint in webstorm, because I am using vscode and had no problem.
> > Adding `ts` files to node extensions is a fix it seems. `eslint-import-resolver-typescript` is used and I thought that does the trick for finding `ts` files. I'll look into this.
> 
> I didn't check because I installed eslint-loader and it works fine.
> However I checked `npm run lint` and it failed on `lint:css` so I changed script line to `"stylelint \"./app/**/*.{js,ts,tsx}\""` but now it fails on `"lint:eslint:quiet` what's strange I have proper errors from es-lint loader in browser and there it's not failing
> ![Imgur](https://camo.githubusercontent.com/5cda9d3f77b9ecafb96b68c7371706349e8a37e0/68747470733a2f2f696d6775722e636f6d2f634775474937622e706e67)

I copied a command from `lint:eslint:quiet` and paste it into the console, there is no error so error above is just exit? 
![image](https://user-images.githubusercontent.com/40576160/78499273-f6811d80-774f-11ea-975f-132544b9b995.png)

**Comment by Surm4** [comment]
Eslint works for me I've had a problem with husky too so I had to reinstall it with cmd run as administrator (windows) so I think adding .ts and .tsx extensions to fix this error is enough

**Comment by Can-Sahin** [comment]
fixed in #71
